File: peuler/ex3/ex3.py
import math

num = 600851475143

def isprime(x):
    for i in range(2, math.ceil(math.sqrt(x))):
        if (x % i == 0):
            return 0
    return 1

def divisors(x):
    divs = []
    for i in range(2, math.ceil(math.sqrt(x))):
        if (x % i == 0):
            divs.append(i)
    return divs

# for some reason, the solution that checks (isprime(i))
# and (num % i == 0) fails (hangs)

for i in sorted(divisors(num), reverse=True):
    if (isprime(i)):
        print("The largest prime divisor of " + str(num) + " is " + str(i))
        break

File: peuler/ex3/ex3.c
#include<stdio.h>
#include<math.h>

#define NUMBER 600851475143

long isprime(long x) {
  for (long i = 2; i <= (int) sqrt(x); i++)
	if (x % i == 0)
	  return 0;
  return 1;
}

int main() {
  unsigned long divisor = (long) sqrt(NUMBER);

  for (int i = (int) sqrt(NUMBER); i >= 2; i--)
	if (isprime(i) && (NUMBER % i == 0)) {
	  printf("The largest prime factor of %lu is %lu.\n", NUMBER, i);
	  break;
	}
  
  return 0;
}
